test/v3/createPool/weighted.integration.test.ts

```diff
@@ -0,0 +1,172 @@
+// pnpm test -- createPool/weighted.integration.test.ts
+
+import {
+    Address,
+    createTestClient,
+    http,
+    parseEther,
+    publicActions,
+    walletActions,
+    zeroAddress,
+} from 'viem';
+import { CHAINS, ChainId, PoolType, TokenType } from '../../../src';
+import { CreatePool } from '../../../src/entities/createPool';
+import { ANVIL_NETWORKS, startFork } from '../../anvil/anvil-global-setup';
+import { doCreatePool } from '../../lib/utils/createPoolHelper';
+import { CreatePoolTxInput } from '../../lib/utils/types';
+import {
+    CreatePoolInput,
+    CreatePoolV3WeightedInput,
+} from '../../../src/entities/createPool/types';
+import { TOKENS } from 'test/lib/utils/addresses';
+
+const { rpcUrl } = await startFork(ANVIL_NETWORKS.SEPOLIA, undefined, 5247812n);
+
+describe('Create Weighted Pool tests', () => {
+    const chainId = ChainId.SEPOLIA;
+    let txInput: CreatePoolTxInput;
+    let poolAddress: Address;
+    let createWeightedPoolInput: CreatePoolV3WeightedInput;
+    beforeAll(async () => {
+        const client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+        const signerAddress = (await client.getAddresses())[0];
+
+        txInput = {
+            client,
+            createPool: new CreatePool(),
+            testAddress: signerAddress,
+            createPoolInput: {} as CreatePoolInput,
+        };
+
+        createWeightedPoolInput = {
+            poolType: PoolType.Weighted,
+            symbol: '50BAL-50WETH',
+            tokens: [
+                {
+                    tokenAddress: TOKENS[chainId].BAL.address, // BAL
+                    weight: parseEther(`${1 / 2}`),
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                },
+                {
+                    tokenAddress: TOKENS[chainId].WETH.address, // WETH
+                    weight: parseEther(`${1 / 2}`),
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                },
+            ],
+            balancerVersion: 3,
+        };
+    });
+    test('Create Weighted Pool', async () => {
+        poolAddress = await doCreatePool({
+            ...txInput,
+            createPoolInput: createWeightedPoolInput,
+        });
+        expect(poolAddress).to.not.be.undefined;
+    });
+
+    test('Wrong weights, expects error', async () => {
```

Comment:
Should the following tests be moved to non-integration?

Reply:
Absolutely, thanks! I created a task on the board to also verify and fix this on the V2 Create and Init Pool test files